WordPress/Classes/ViewRelated/Menus/Views/MenuItemAbstractPostsViewController.m

```diff
@@ -0,0 +1,201 @@
+    
+    [self deselectVisibleSourceCellsIfNeeded];
+    
+    MenuItemSourceCell *selectedCell = [tableView cellForRowAtIndexPath:indexPath];
+    selectedCell.sourceSelected = YES;
+}
+
+#pragma mark - MenuItemSourcePostAbstractViewSubclass
+
+- (Class)entityClass
+{
+    AssertSubclassMethod();
+    return nil;
+}
+
+- (PostServiceSyncOptions *)syncOptions
+{
+    PostServiceSyncOptions *options = [[PostServiceSyncOptions alloc] init];
+    options.statuses = @[PostStatusPublish, PostStatusPrivate];
+    options.number = @(PostServiceDefaultNumberToSync);
+    return options;
+}
+
+- (void)didFinishSyncingPosts:(NSArray *)posts options:(PostServiceSyncOptions *)options
+{
+    if (posts) {
+        self.numberOfSyncedPosts = self.numberOfSyncedPosts + posts.count;
+        self.additionalPostsAvailableForSync = posts.count >= options.number.unsignedIntegerValue;
+        self.resultsController.fetchRequest.fetchLimit = self.numberOfSyncedPosts;
+        [self performResultsControllerFetchRequest];
+        [self.tableView reloadData];
+    }
+    [self hideLoadingSourcesIndicator];
+}
+
+#pragma mark - paging
+
+- (void)scrollingWillDisplayEndOfTableView:(UITableView *)tableView
+{
+    if (!self.additionalPostsAvailableForSync || [self searchBarInputIsActive]) {
+        return;
+    }
+    [self showLoadingSourcesIndicator];
+    PostService *service = [[PostService alloc] initWithManagedObjectContext:[self managedObjectContext]];
+    PostServiceSyncOptions *options = [self syncOptions];
+    options.offset = @(self.resultsController.fetchedObjects.count);
+    [service syncPostsOfType:[self sourceItemType]
+                 withOptions:options
+                     forBlog:[self blog]
+                     success:^(NSArray *posts) {
+                         [self didFinishSyncingPosts:posts options:options];
+                     } failure:^(NSError *error) {
+                         [self didFinishSyncingPosts:nil options:options];
+                         [self showLoadingErrorMessageForResults];
+                     }];
+}
+
+#pragma mark - searching
+
+- (void)searchBarInputChangeDetectedForLocalResultsUpdateWithText:(NSString *)searchText
+{
+    NSPredicate *defaultPredicate = [self defaultFetchRequestPredicate];
+    NSPredicate *predicate = nil;
+    
+    if (searchText.length) {
+        NSPredicate *searchPredicate = [NSPredicate predicateWithFormat:@"postTitle CONTAINS[cd] %@", searchText];
+        predicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[defaultPredicate, searchPredicate]];
+        if ([self.resultsController.fetchRequest.predicate isEqual:predicate]) {
+            // same predicate, no update needed
+            return;
+        }
+        if (self.additionalPostsAvailableForSync) {
+            self.defersFooterViewMessageUpdates = YES;
+        }
+    } else {
+        predicate = defaultPredicate;
+    }
+    
+    DDLogDebug(@"MenuItemSourcePostView: Updating fetch request predicate");
```

Comment:
Since we're logging for debug, it might be interesting to know if it was the default, or if `searchText` had been specified.